From: David Milburn <dmilburn@redhat.com>
Date: Wed, 18 Aug 2010 23:59:39 -0400
Subject: [ata] sata_mv: enable async_notify for 60x1 Rev.C0 and up
Message-id: <1282175987-21696-2-git-send-email-dmilburn@redhat.com>
Patchwork-id: 27692
O-Subject: [RHEL5.6 PATCH 1/9] BZ 554872 sata_mv: enable async_notify for 60x1
	Rev.C0 and higher
Bugzilla: 554872
RH-Acked-by: Jeff Garzik <jgarzik@redhat.com>

commit 3bd0a70ee9cc30ae81b39cb5ecad0fa7bcb4675b
Author: Mark Lord <liml@rtr.ca>
Date:   Wed Jun 18 12:11:16 2008 -0400

    sata_mv: enable async_notify for 60x1 Rev.C0 and higher

BZ 554872

diff --git a/drivers/ata/sata_mv.c b/drivers/ata/sata_mv.c
index 36a2090..fe71321 100644
--- a/drivers/ata/sata_mv.c
+++ b/drivers/ata/sata_mv.c
@@ -1322,6 +1322,9 @@ static int mv_port_start(struct ata_port *ap)
 		goto out_port_free_dma_mem;
 	memset(pp->crpb, 0, MV_CRPB_Q_SZ);
 
+	/* 6041/6081 Rev. "C0" (and newer) are okay with async notify */
+	if (hpriv->hp_flags & MV_HP_ERRATA_60X1C0)
+		ap->flags |= ATA_FLAG_AN;
 	/*
 	 * For GEN_I, there's no NCQ, so we only allocate a single sg_tbl.
 	 * For later hardware, we need one unique sg_tbl per NCQ tag.
